scripts/prepare_data.py

```diff
@@ -62,67 +164,120 @@ def add_tile_id(row):
     # Prepare the tiles
 
     ## Convert datasets shapefiles into geojson format
-    logger.info('Convert label shapefile into GeoJSON format (EPSG:4326)...')
-    labels = gpd.read_file(SHPFILE)
-    labels_4326 = labels.to_crs(epsg=4326)
-
-    nb_labels = len(labels)
-    logger.info('There is/are ' + str(nb_labels) + ' polygon(s) in ' + SHPFILE)
+    logger.info('Convert labels shapefile into GeoJSON format (EPSG:4326)...')
+    labels_gdf = gpd.read_file(SHPFILE)
+    if 'year' in labels_gdf.keys():
+        labels_gdf['year'] = labels_gdf.year.astype(int)
+        labels_4326_gdf = labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry', 'year'])
+    else:
+        labels_4326_gdf = labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry'])
+    nb_labels = len(labels_gdf)
+    logger.info(f'There are {nb_labels} polygons in {SHPFILE}')
+
+    labels_4326_gdf['CATEGORY'] = 'mineral extraction site'
+    labels_4326_gdf['SUPERCATEGORY'] = 'land usage'
+    
+    gt_labels_4326_gdf = labels_4326_gdf.copy()
 
     label_filename = 'labels.geojson'
     label_filepath = os.path.join(OUTPUT_DIR, label_filename)
-    labels_4326.to_file(label_filepath, driver='GeoJSON')
+    labels_4326_gdf.to_file(label_filepath, driver='GeoJSON')
     written_files.append(label_filepath)  
     logger.success(f"{DONE_MSG} A file was written: {label_filepath}")
 
-    logger.info('Creating tiles for the Area of Interest (AoI)...')   
-    
-    # Grid definition
-    tms = morecantile.tms.get("WebMercatorQuad")    # epsg:3857
-
-    # New gpd with only labels geometric info (minx, miny, maxx, maxy) 
-    logger.info('- Get geometric boundaries of the label(s)')  
-    label_boundaries_df = labels_4326.bounds
-
-    # Iterate on geometric coordinates to defined tiles for a given label at a given zoom level
-    # A gpd is created for each label and are then concatenate into a single gpd 
-    logger.info('- Compute tiles for each label geometry') 
-    tiles_4326_all = [] 
+    # Add FP labels if it exists
+    if FP_SHPFILE:
+        fp_labels_gdf = gpd.read_file(FP_SHPFILE)
+        assert_year(fp_labels_gdf, labels_gdf, 'FP', EPT_YEAR) 
+        if 'year' in fp_labels_gdf.keys():
+            fp_labels_gdf['year'] = fp_labels_gdf.year.astype(int)
+            fp_labels_4326_gdf = fp_labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry', 'year'])
+        else:
+            fp_labels_4326_gdf = fp_labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry'])
+        fp_labels_4326_gdf['CATEGORY'] = 'mineral extraction site'
+        fp_labels_4326_gdf['SUPERCATEGORY'] =  'land usage'
+
+        nb_fp_labels = len(fp_labels_gdf)
+        logger.info(f"There are {nb_fp_labels} polygons in {FP_SHPFILE}")
+
+        filename = 'FP.geojson'
```

Comment:
This variable could be avoided by just writing the filename directly in the file path.